scripts/build/store-azp-variables.ps1

```diff
@@ -0,0 +1,15 @@
+# Read the project version from the Version.props file
+$VersionFilePath = "$env:BUILD_SOURCESDIRECTORY\scripts\version\Version.props"
+Write-Host "Reading the Sonar project version from '${VersionFilePath}' ..."
+
+[xml]$VersionProps = Get-Content "$VersionFilePath"
+$Version = $VersionProps.Project.PropertyGroup.MainVersion
+Write-Host "Sonar project version is '${Version}'"
+
+# Save variables to files so they can be used by other tasks and stages
+$Dir = "$env:AGENT_BUILDDIRECTORY\azp-variables"
```

Comment:
`AGENT_BUILDDIRECTORY`=`PIPELINE_WORKSPACE`
`BUILD_SOURCESDIRECTORY`=`PIPELINE_WORKSPACE` + `\s`

I'm putting this outside sources directory, to avoid the cleanup of `Post-job: Checkout SonarSource/sonar-dotnet@master to s` job.

Docs: https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml